packages/js-commons/src/core/local-files.ts

```diff
@@ -97,70 +103,173 @@ export const getLocalApplicationTransporter = (
       path?: string,
     ): Promise<ApplicationDTO> => {
       ensureDir(ensembleDir);
+      const updatedAppData = omit(appData, ["assets", "fonts"]);
       const appsMetaData = await getGlobalMetadata();
       const existingAppMetadata = (
         appsMetaData[appData.id]
-          ? { projectPath: path, ...appsMetaData[appData.id], ...appData }
-          : { ...appData, projectPath: path }
+          ? {
+              projectPath: path,
+              ...appsMetaData[appData.id],
+              ...updatedAppData,
+            }
+          : { ...updatedAppData, projectPath: path }
       ) as ApplicationLocalMeta;
       if (path) {
         existingAppMetadata.projectPath = path;
       } else {
         existingAppMetadata.projectPath = join(ensembleDir, appData.id);
       }
 
-      const documentsToWrite = values(
-        pick(appData, [
-          "screens",
-          "widgets",
-          "scripts",
-          "assets",
-          "translations",
-          "env",
-          "theme",
-        ]),
-      ).flatMap((docset: unknown) => {
-        if (isArray(docset)) {
-          return docset as EnsembleDocument[];
-        }
-        if (!isNil(docset)) {
-          return [docset as EnsembleDocument];
-        }
-        return [];
-      });
+      const documentsToWrite = values(pick(appData, ArtifactProps)).flatMap(
+        (docset: unknown) => {
+          if (isArray(docset)) {
+            return docset as EnsembleDocument[];
+          }
+          if (!isNil(docset)) {
+            return [docset as EnsembleDocument];
+          }
+          return [];
+        },
+      );
 
       const yamlFileWrites = compact(documentsToWrite).map(async (document) => {
-        const { relativePath } = await saveArtifact(
-          document,
-          existingAppMetadata,
-          {
+        let relativePath = "";
+        let docToWrite = document;
+
+        if (isAssetOrFont(document)) {
+          const asset = document as AssetDTO | FontDTO;
+          if (!asset.publicUrl) return;
+
+          const fileData = await fetchFileData(asset.publicUrl);
```

Comment:
You will need to consider that sometimes the asset needs to be fetched from the local file system, and not the publicUrl.